randomteleport-plugin/src/main/java/de/themoep/randomteleport/RandomTeleport.java

```diff
@@ -364,4 +367,19 @@ public RandomSearcher runPreset(CommandSender sender, String preset, Player targ
         }
         return parseAndRun(sender, center, cmd.split(" "));
     }
+
+    @Override
+    public Location getRandomLocation(Player player, Location origin, int minRange, int maxRange, LocationValidator... validators) throws ExecutionException, InterruptedException {
+        return getRandomSearcher(player, origin, minRange, maxRange, validators).search().get();
```

Comment:
This will block the thread that the method was called from, this should either be noted in the docs or even better: Return the CompletableFuture so a dev can decide for themselves what he want to do.

Reply:
Updated that should now return a CompletableFuture<Location>